tests/Assistants/AssistantsTests.cs

```diff
@@ -847,6 +852,45 @@ This file describes the favorite foods of several people.
         });
     }
 
+    [Test]
+    public async Task FileOnMessageWorks()
+    {
+        // First, we need to upload a simple test file.
+        OpenAIFileClient fileClient = GetTestClient<OpenAIFileClient>(TestScenario.Files);
+        OpenAIFile testFile = fileClient.UploadFile(
+            BinaryData.FromString("""
+            This file describes the favorite foods of several people.
+
+            Summanus Ferdinand: tacos
+            Tekakwitha Effie: pizza
+            Filip Carola: cake
+            """).ToStream(),
+            "favorite_foods.txt",
+            FileUploadPurpose.Assistants);
+        Validate(testFile);
+
+        AssistantClient client = GetTestClient();
+
+        var thread = client.CreateThread();
+        var assistant = client.CreateAssistant("gpt-4o-mini");
```

Comment:
1. Switch to the async version of this method: `CreateAssistantAsync`
2. Then call `Validate(assistant);`